adding recoveries XML to simulation launch file

SteveMacenski opened the pull request:
Adding the `..._and_recovery` XML file for the simulation launch file so we can have the recovery behaviors running in the simulation. There were a few issues I resolved that likely would have been seen earlier if it were default.

orduno commented:
@SteveMacenski Before merging, could do a simple test running navigation (with the new default bt) and triggering a recovery. @crdelsey mentioned the collision checker tests had to be disabled for master to pass CI, could be that recoveries are not performed correctly in which case we might not want to set them as default for now.

SteveMacenski commented:
@orduno see video attached. I'm pretty confident as far as I can tell that it works fine.

Things to note:
- When in collision with stuff, spinning stopped
- When that costmap cell went away and I spun again, it spun until it came back and stopped immediately
- When I moved the robot into free space with low cost, spun fine
- When I moved it again a solid wall it stopped 
- When the footprint clearing cleared the cost in contact with the wall and the robot moved over slightly, that edge of costmap cell triggered it to stop

That seems to cover all the possible edge cases I can think of

![ezgif com-video-to-gif(1)](https://user-images.githubusercontent.com/14944147/62902258-f2f9db80-bd13-11e9-9b44-cefeaba374ba.gif)

orduno commented:
Thanks @SteveMacenski
